iib/workers/tasks/build.py

```diff
@@ -310,8 +311,13 @@ def _get_resolved_image(pull_spec):
     :rtype: str
     """
     log.debug('Resolving %s', pull_spec)
-    skopeo_output = skopeo_inspect(f'docker://{pull_spec}')
-    pull_spec_resolved = f'{skopeo_output["Name"]}@{skopeo_output["Digest"]}'
+    if '@' in pull_spec:
+        name = pull_spec.rsplit('@', 1)[0]
+    else:
+        name = pull_spec.rsplit(':', 1)[0]
```

Comment:
This `if-else` block is used at least 2 more times in the same file. Do you think it should be a helper method instead?

Reply:
I added a new commit which moves this logic to a separate function.